pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +81,27 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := watchForFailedPushEvent(eventClient, ref.Namespace, ref.Name)
```

Comment:
If you can't make this decision with the info we have in the builder already, then we didn't do quota right.